lms/djangoapps/grades/tests/test_tasks.py

```diff
@@ -170,17 +191,22 @@ def test_query_count_does_not_change_with_more_problems(self, default_store, add
             self.assertTrue(PersistentGradesEnabledFlag.feature_enabled(self.course.id))
             ItemFactory.create(parent=self.sequential, category='problem', display_name='problem2')
             ItemFactory.create(parent=self.sequential, category='problem', display_name='problem3')
-            with check_mongo_calls(2) and self.assertNumQueries(25 + added_queries):
-                recalculate_subsection_grade.apply(args=tuple(self.score_changed_kwargs.values()))
+            with self.mock_student_module():
+                with check_mongo_calls(2) and self.assertNumQueries(25 + added_queries):
+                    recalculate_subsection_grade.apply(args=tuple(self.score_changed_kwargs.values()))
```

Comment:
To DRY this and  to make it more readable, add a helper function that calls `recalculate_subsection_grade`.  And move all the calls to `self.mock_student_module()` within the helper function.  (Except for the call in `test_signal_queues_task`.)